libopendroneid/wifi.c

```diff
@@ -0,0 +1,340 @@
+/*
+Copyright (C) 2019 Intel Corporation
+
+SPDX-License-Identifier: Apache-2.0
+
+Open Drone ID C Library
+
+Maintainer:
+Gabriel Cox
+gabriel.c.cox@intel.com
+*/
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <errno.h>
+#include <byteswap.h>
+
+#include "opendroneid.h"
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define cpu_to_le16(x)  (x)
+#else
+#define cpu_to_le16(x)      (bswap_16(x))
+#endif
+
+#define IEEE80211_FCTL_FTYPE          0x000c
+#define IEEE80211_FCTL_STYPE          0x00f0
+
+#define IEEE80211_FTYPE_MGMT            0x0000
+#define IEEE80211_STYPE_ACTION          0x00D0
+
+
+char *drone_export_gps_data(ODID_UAS_Data *UAS_Data)
+{
+	int len = 0, total_len = 8192;
+	char *drone_str;
+
+	drone_str = malloc(total_len);
+	if (!drone_str)
+		return NULL;
+
+#define mprintf(...) {\
+	if (total_len - len < 2048) { \
+		total_len *= 2; \
+		drone_str = realloc(drone_str, total_len); \
+		if (!drone_str) \
+			return NULL; \
+	} \
+	len += snprintf(drone_str + len, total_len - len, __VA_ARGS__); \
+	if (len > total_len) { \
+		free(drone_str); \
+		return NULL; \
+	} \
+}
+	/* build json object from UAS_Data */
+	/*
+	{
+		"Version": "x.x",
+		"Response": {
```

Comment:
I am not very familiar with the usage of the json output format (I guess it relates to the Remote Network ID primarily?). However, the below is not completely in line with the message definitions in v0.8 of the specification:

Location: Missing height type bit, baro altitude accuracy
Authentication: Missing length (bytes) and timestamp
System/Operator: Missing operator location type, area floor
Operator ID: This message is missing completely.

Reply:
@friissoren can you please point me to the right json format? The specs still have version 0.64.3:

https://github.com/opendroneid/specs